index.js

```diff
@@ -1,11 +1,30 @@
 var path = require('path');
+var fs = require('fs');
 var findParentDir = require('find-parent-dir');
 
+/**
+ * Resolves the url in the top level node_module directory
+ * @param {String} targetUrl url to resolve
+ * @param {String} source file or directory which is the base point resolving targetUrl
+ */
+function resolve(targetUrl, source) {
+  var packageRoot = findParentDir.sync(source, 'node_modules');
+
+  if (!packageRoot) {
+    return null; // file could not be resolved as npm package
+  }
+
+  var filePath = path.resolve(packageRoot, 'node_modules', targetUrl);
+
+  return fs.existsSync(path.dirname(filePath))
+    ? filePath
+    : resolve(targetUrl, path.dirname(packageRoot));
+}
+
 module.exports = function importer (url, prev, done) {
   if (url[ 0 ] === '~') {
-    var packageRoot = findParentDir.sync(prev, 'node_modules') || __dirname;
-    url = path.resolve(packageRoot, 'node_modules', url.substr(1));
+    url = resolve(url.substr(1), prev);
   }
 
-  return { file: url };
+  return url && { file: url };
```

Comment:
I think we always need to return an object literal, se https://github.com/sass/node-sass#importer--v200---experimental